### PR TITLE
validators: enable multi-process segment validation

### LINKED ISSUE #182: validation: make the validation file process aware
Right now the validation figures out the schema that should be applied only from the action name. It should also use the process name of the segment.

### PR DESCRIPTION
**Pull request by simonvadee**
Resolves #182

### CODE REVIEW

**Comment by simonvadee** [comment]
In the last commit I updated TestRootValidator to cover the 5% that was missing due to the previous commit.

File: validator/root.go [hunk 1]
```diff
@@ -50,7 +50,12 @@ func NewRootValidator(filename string, validByDefault bool) Validator {
 
 func (rv rootValidator) Validate(store store.Reader, segment *cs.Segment) error {
 	validByDefault := rv.ValidByDefault
-	for _, validator := range rv.Validators {
+	actionValidators, exists := rv.ProcessValidators[segment.Link.GetProcess()]
+	if !exists && !validByDefault {
+		return errors.New("root validation failed : process validation not found")
```
Comment (such) [inline validator/root.go]: typo: no space before a column in english

File: validator/root.go [hunk 2]
```diff
@@ -50,7 +50,12 @@ func NewRootValidator(filename string, validByDefault bool) Validator {
 
 func (rv rootValidator) Validate(store store.Reader, segment *cs.Segment) error {
 	validByDefault := rv.ValidByDefault
-	for _, validator := range rv.Validators {
+	actionValidators, exists := rv.ProcessValidators[segment.Link.GetProcess()]
```
Comment (such) [inline validator/root.go]: shouldn't they be called processValidators?

File: validator/root.go [hunk 3]
```diff
@@ -14,16 +14,16 @@ import (
 )
 
 type rootValidator struct {
-	ValidByDefault bool
-	Validators     []selectiveValidator
+	ValidByDefault    bool
+	ProcessValidators map[string][]selectiveValidator
```
Comment (such) [inline validator/root.go]: I'd call this ValidatorsByProcess

**Comment by such** [comment]
Great!
Could you check what needs to be done on the `generators` side? I think changes related to validators are on the 0.1.0-dev branch

**Comment by simonvadee** [comment]
alright

**Pull request merged**